### PR TITLE
create DM delay code and tests

### PR DESCRIPTION
**Pull request by eflewis312**
Added DM delay calculation as a function in arts/observation.py. Center frequencies/bandwidths need to be in MHz for this to work properly; not sure if there's a good way to make sure this is always the case.

Added test cases to tests/test_observation.py. Using pytest.approx to make sure DM delay is 0 when bandwidth/DM are 0, and checks math against one test case.

### CODE REVIEW

File: arts/observation.py [hunk 1]
```diff
@@ -34,12 +34,26 @@ def dm_delay(self, dm):
...
+        #Make sure frequencies are in MHz otherwise this is wrong
+
+        chan_width = self.bandwidth/self.nchans
+        lowchan_end = lf + chan_width #upper frequency of lowest-frequency channel
+        hichan_start = hf - chan_width #lower frequency of highest-frequency channel
+
+        t_delay_bw = 4148808 * dm * ( (lf**(-2)) - (hf**(-2)) ) #delay across BW in ms
+        t_delay_lowchan = 4148808 * dm * ( (lf**(-2)) - (lowchan_end**(-2)) ) #delay across lowest channel in ms
+        t_delay_hichan = 4148808 * dm * ( (hichan_start**(-2)) - (hf**(-2)) ) #delay across highest channel in ms
+        return t_delay_bw, t_delay_lowchan, t_delay_hichan
```
Comment (KshitijAggarwal) [inline arts/observation.py:56]: It would be useful to add an input to this function that decides which of the three delays to calculate and return

**Comment by KshitijAggarwal** [comment]
If you haven't already, please reformat your code using [black](https://github.com/psf/black).

**Comment by eflewis312** [comment]
Code has been reformatted to use black; added argument to determine which DM delay gets returned (default is across entire bandwidth); and added test to check.

**Comment by KshitijAggarwal** [comment]
Thanks for the changes. Merging it.

**Pull request merged**